### PR TITLE
Avoid some Json copy operations.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by axic** [comment]
Do you want to do the same to `StandardCompiler` and `jsonCompiler` in this PR or a separate one?

File: libsolidity/ast/ASTJsonConverter.h [hunk 1]
```diff
@@ -156,6 +159,11 @@ class ASTJsonConverter: public ASTConstVisitor
 		std::vector<std::pair<std::string, Json::Value>> &_attributes,
 		ExpressionAnnotation const& _annotation
 	);
+	static void appendMove(Json::Value& _array, Json::Value&& _value)
+	{
+		_array[_array.size()] = std::move(_value);
```
Comment (axic) [inline libsolidity/ast/ASTJsonConverter.h]: An assert here that `_array` is of `arrayValue` type?

**Review by axic — APPROVED** [review]

**Comment by chriseth** [comment]
Ok, I'm lost why this is failing so many tests...

File: libsolidity/ast/ASTJsonConverter.h [hunk 2]
```diff
@@ -155,6 +158,12 @@ class ASTJsonConverter: public ASTConstVisitor
 		std::vector<std::pair<std::string, Json::Value>> &_attributes,
 		ExpressionAnnotation const& _annotation
 	);
+	static void appendMove(Json::Value& _array, Json::Value&& _value)
+	{
+		solAssert(_array.type() == Json::arrayValue, "");
```
Comment (axic) [inline libsolidity/ast/ASTJsonConverter.h]: Could use `isArray()`.

File: libsolidity/ast/ASTJsonConverter.h [hunk 3]
```diff
@@ -155,6 +158,12 @@ class ASTJsonConverter: public ASTConstVisitor
 		std::vector<std::pair<std::string, Json::Value>> &_attributes,
 		ExpressionAnnotation const& _annotation
 	);
+	static void appendMove(Json::Value& _array, Json::Value&& _value)
+	{
+		solAssert(_array.isArray(), "");
```
Comment (axic) [inline libsolidity/ast/ASTJsonConverter.h:163]: This is explicit now and hence we need to create the "children" arrays above, however if we remove the assert here, jsoncpp will internally just created it on `append`. Not sure which is faster, but this is more explicit that's for sure.

Comment (chriseth) [reply libsolidity/ast/ASTJsonConverter.h:163]: Thanks!

Comment (axic) [reply libsolidity/ast/ASTJsonConverter.h:163]: @chriseth are we merging this version then?

**Pull request merged**